### ISSUE TITLE
Bootstrap 4 wrapper 

### ISSUE THREAD

**Issue by Starzu** [UdashFramework/udash-core#205]

**Comment by sbrunk** [comment]
We're looking into using Udash for a new project as it looks like just what we need. We'd like to use Bootstrap 4 though, if possible. So I have a few questions:

Is this something you plan to add soon?
Could I help you out here in some way?
I stumpled upon https://github.com/Karasiq/scalajs-bootstrap which supports Bootstrap 4 and uses ScalaTags as well. So I'm wondering if it could make sense to reuse some parts of it or if it's too different.

Thanks in advance.

**Comment by sbrunk** [comment]
Thanks @Starzu I'm going to look into it.

**Comment by Starzu** [comment]
How is it going @sbrunk? I saw you've migrated at least a part of this.

**Comment by sbrunk** [comment]
Yes I'm still on it although I was busy with other stuff this week. I hope I can continue to work on it next week.

Just in case you're faster than me and you want to take over, feel free to build on my branch: https://github.com/sbrunk/udash-core/tree/bootstrap4

**Comment by sbrunk** [comment]
Since you seem to be faster, here's what I found out about the date picker component: The current [date picker](https://github.com/Eonasdan/bootstrap-datetimepicker) is only compatible with BS 3.

The author [refers](https://github.com/Eonasdan/bootstrap-datetimepicker/issues/1377#issuecomment-309788107) https://github.com/tempusdominus/bootstrap-4 as an alternative for BS 4, which seems to be kind of the successor (the versioning is a bit confusing for me).

**Comment by Starzu** [comment]
Thanks, I'll check it.

**Comment by Starzu** [comment]
Hi @sbrunk. I'm starting my vacation tomorrow. I've not finished the wrapper, but I've pushed all my changes on the `bootstrap4` branch. It contains all Bootstrap 4 components (datepicker's events are broken - https://github.com/tempusdominus/bootstrap-4/issues/190 - but everything else looks good), so I belive it's usable and wont drastically change from this moment. 

The API is not compatible with the Bootstrap 3 wrapper. I have introduced similar changes to the changes in the core `InputBinding`s:
 * Each component extends `Binding` interface, so you can call `kill()` on it to cancel all properties listeners. 
 * Factory methods have two arguments lists: behavior/settings/data and factories of subelements.
 * Factories provide `Binding.NestedInterceptor` as an argument.

**Comment by sbrunk** [comment]
Thanks @Starzu. I just came back from vacation myself. I'll give it a try next week.
